internal/k8s/k8s.go

```diff
@@ -68,17 +72,18 @@ const (
 // Config specifies the configuration of the Kubernetes
 // client/watcher.
 type Config struct {
-	ProcessName   string
-	ConfigMapName string
-	ConfigMapNS   string
-	NodeName      string
-	MetricsHost   string
-	MetricsPort   int
-	ReadEndpoints bool
-	Logger        log.Logger
-	Kubeconfig    string
-
-	ServiceChanged func(log.Logger, string, *v1.Service, *v1.Endpoints) SyncState
+	ProcessName        string
+	ConfigMapName      string
+	ConfigMapNS        string
+	NodeName           string
+	MetricsHost        string
+	MetricsPort        int
+	ReadEndpoints      bool
```

Comment:
Any idea why we would not "ReadEndpoints"?

Reply:
That was already there, and the reason is because the same k8s client is used by the controller that doesn't need it:
https://github.com/metallb/metallb/blob/0c83b4ba1164ba27a64084a6b0bde12a38c2d24a/controller/main.go#L153